### DISCUSSION TITLE
Keine Verbindung OpenDTU zum Wechselrichter, dieser ist aber an

### DISCUSSION THREAD

**Discussion by z-master42** [tbnobody/OpenDTU#1946] [Q&A]
Hallo, ich habe folgendes Phänomen bei meiner Installation. Ich habe zwei Hoymiles HM-600 in der Nutzung, welche normal miteinander verbunden sind und einen gemeinsamen Anschluss nutzen. Soweit so gut. An einem Wechselrichter hängt ein SolarFlow PV-Hub, an dem anderen direkt Module. Der Wechselrichter mit den Modulen ist ungeregelt, den mit dem PV-Hub regele ich verbrauchsabhängig über die OpenDTU. So kommt es gerade morgens zu der Konstellation, dass ein Wechselrichter ein (der mit den direkt angeschlossenen Modulen) ist und einer (der mit dem PV-Hub) aus. Die OpenDTU schafft es hierbei aber nicht, eine Verbindung zu dem einen laufenden Wechselrichter herzustellen. Erst wenn beide Wechselrichter einspeisen steht auch die Verbindung über die OpenDTU. 
Jemand eine Idee woran dies liegen könnte?
Muss ich hierfür in der OpenDTU speziell was einstellen?
Ich konnte noch nicht kontrollieren ob bei dem laufenden Wechselrichter die LED grün alle 4 s blinkt, was ja anzeigt, dass keine Verbindung zur DTU besteht.

Problem hierbei ist natürlich, dass ich solange auch keine Werte per MQTT bekomme.

Wenn ich mich nicht verguckt habe, besteht das Problem andersherum allerdings nicht. Also wenn abends keine Sonne mehr vorhanden ist und der PV-Hub Wechselrichter aus den Batterien gespeist wird.

**Comment by z-master42** [comment]
Update: Habe jetzt alle mir möglichen Konstellationen durch. Der Fehler bleibt der gleiche: Wenn Wechselrichter 1 aus ist, ist Wechselrichter 2 in der OpenDTU (ebenfalls) nicht erreichbar, obwohl er an ist und produziert (LED blinkt grün alle 1 s). Schalte ich den Wechselrichter 1 ein, sind beide ganz normal erreichbar.

| | Wechselrichter 1 | Wechselrichter 2| 
| ---| ---|--- | 
|Modell|HM-600-2T|HM-600-2T|
|Produktionsdatum |50/2022 |40/2022 | 
|Firmware-Version|1.0.16 | 1.0.10| 
|Hardware-Version|01.10| 01.10|

**Reply by OttoKarlHeinz** [reply]
@z-master42 hat sich das Problem gelöst? Bei mir tritt ein änhliches Problem abends mit dem Sonnenaufgang auf siehe #2269

**Reply by z-master42** [reply]
Es liegt tatsächlich an dem gleichen Problem wie in deinem Issue. Wenn der Wechselrichter der per OpenDTU für eine Nulleinspeisung gesteuert werden soll gar nicht an ist, sammeln sich die Steuerungsbefehle "in der DTU" und sie müllt quasi voll, sodass die nicht mehr mit dem anderen Wechselrichter reden kann. Ich habe es dadurch gelöst, dass ich meine Nulleinspeisungsteuerung (über Home Assistant und MQTT) entsprechend verfeinert habe, was den Status des Wechselrichters angeht. Stichwort: Erreichbar und Produzierend.

**Reply by OttoKarlHeinz** [reply]
Ja die OpenDTU braucht ca. 2 Minuten um die MQTT Steuerungsbefehle von  der Nulleinspeisung des HA an den WR 2  zu senden. Ich probiere das mal mit producing off, dann gehen keine Steuerungsbefehle mehr an die OpenDTU.
Ähnliches mache ich mit dem WR 2 (von Batterie gespeist), der wird ausgeschaltet, wenn WR 1  (ohne Batterie) ausreichend Leistung für die Nulleinspeisung zur Verfügung stellen kann und bekommt dann keine Steuerungsbefehle mehr.
Sauber ist die Programmierung in der OpenDTU an dieser Stelle aber nicht. Im Prinzip müssten die MQTT- Steuerungsbefehle bei producing off ignoriert und damit  die zeitlich verzögerte Abarbeitung der Steuerungsbefehle verhindert  werden.

**Reply by OttoKarlHeinz** [reply]
Hallo z-master42, die Steuerung über producing "off" löst das Problem. Danke schön.

**Reply by mazzhats** [reply]
Hi. Könnte mir einer erklären, was ihr genau meint. Und wie ihr dieses producing off umgesetzt habt?
Gruss

**Reply by z-master42** [reply]
Ich meine damit, dass ich in meiner Home Assistant Automatisierung nicht nur darauf gucke, ob der Wechselrichter erreichbar ist, sondern ob er auch noch produziert. Die OpenDTU liefert beide Werte per MQTT.

**Comment by mazzhats** [comment]
Wenn ich die Update Zeit der homeassistant Steuerung an die dtu unter 5 Sekunden setze, dann friert die dtu ein.  Erst wenn ich die Progeammierung kurz stoppen und alle Befehle abgearbeitet sind, dann liefert die dtu wieder Werte der Hoymiles wechselrichter.

**Reply by z-master42** [reply]
Ja, das kommt noch dazu.
